src/Servers/Kestrel/Core/src/Internal/Http2/Http2Stream.cs

```diff
@@ -318,8 +318,8 @@ private bool TryValidatePath(ReadOnlySpan<char> pathSegment)
             try
             {
                 // The decoder operates only on raw bytes
-                var pathBuffer = new byte[pathSegment.Length].AsSpan();
-                for (int i = 0; i < pathSegment.Length; i++)
+                Span<byte> pathBuffer = stackalloc byte[pathSegment.Length];
```

Comment:
Should there be a maximum allowed stackalloc size? e.g.

```cs
Span<byte> pathBuffer = pathSegment.Length <= 1024 ? stackalloc byte[pathSegment.Length] : new byte[pathSegment.Length];
```

Reply:
Yes, especially when `pathSegment.Length` can be of any size, otherwise there may be a stackoverflow.

Perf-wise the JIT generates best code by using a constant:
```c#
Span<byte> pathBuffer = pathSegment.Length <= 1024
    ? stackalloc byte[1024].Slice(0, pathSegment.Length)
    : new byte[pathSegment.Length];
```
That's what I meant in my previous comment. It has nothing to do with processing in chunks. It's just about better / easier work for the JIT, as it knows how to stackalloc and doesn't need to emit logic for the allocation depending on the actual size.

`1024` seems pretty huge to me (and afaik this value isn't used in the runtime-repo), I would go with `256` (to be on the safe side with regards to SO-exceptions).

If `pathSegment.Length` will be quite often above 256, also use the array-pool:
```c#
byte[] bufferToReturnToPool = null;

Span<byte> pathBuffer = (pathSegment.Length <= 256
    ? stackalloc byte[256]
    : bufferToReturnToPool = ArrayPool<byte>.Shared.Rent(pathSegment.Length)
).Slice(0, pathSegment.Length);

try
{
    // ...
}
finally
{
    if (bufferToReturnToPool!=null)
    {
        bufferToReturnToPool.AsSpan(0, pathSegment.Length).Clear(); // optional
        ArrayPool<byte>.Shared.Return(bufferToReturnToPool);
    }
}
```

Reply:
The request has size limits, but they may still be larger than you'd want to stack alloc.

Reply:
I imagine that 256 will cover the vast majority of URLs